Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -128,8 +128,20 @@ export class ArtifactEngine {
                     return value;
                 });
 
-                this.artifactItemStore.addItems(items);
-                this.artifactItemStore.updateState(item, models.TicketState.Processed);
+                if (items.length > 0) {
+                    this.artifactItemStore.addItems(items);
+                    this.artifactItemStore.updateState(item, models.TicketState.Processed);
+                }
+                else {
+                    destProvider.putArtifactItem(item, null)
```

Comment:
add a UT for artifact engine too?

Reply:
will take it up in next PR along with e2e tests